### PR TITLE
PHPC-440 and PHPC-465: Abstract WriteException and remove single write methods

### PR DESCRIPTION
**Pull request by jmikola**
https://jira.mongodb.org/browse/PHPC-440

This makes WriteException a common base class of WriteConcernException and WriteErrorException. BulkWriteException, which may contain multiple write and/or write concern errors, will continue to inherit WriteException.

----

https://jira.mongodb.org/browse/PHPC-465

This removes the single write methods on Manager. Additionally, WriteErrorException and WriteConcernException are now unused and also removed.

### CODE REVIEW

**Comment by derickr** [comment]
Tests need to be added for this, otherwise, LGTM.

**Comment by jmikola** [comment]
WriteErrorException and WriteConcernException still need to be removed. I'll get to that shortly.

File: tests/manager/manager-executeBulkWrite-002.phpt [hunk 1]
```diff
@@ -1,5 +1,5 @@
 --TEST--
-MongoDB\Driver\Manager::executeBulkWrite() with duplicate key errors (ordered)
+MongoDB\Driver\Manager::executeBulkWrite() with upserted ids
```
Comment (derickr) [inline tests/manager/manager-executeBulkWrite-002.phpt]: Why did you change this so massively?

File: tests/server/server-executeQuery-004.phpt [hunk 2]
```diff
@@ -7,11 +7,14 @@ MongoDB\Driver\Manager::executeDelete() multiple documents
 require_once __DIR__ . "/../utils/basic.inc";
 
 $manager = new MongoDB\Driver\Manager(STANDALONE);
+$server = $manager->executeQuery(NS, new MongoDB\Driver\Query(array()))->getServer();
```
Comment (derickr) [inline tests/server/server-executeQuery-004.phpt]: Why the addition of $server?

File: tests/replicaset/writeconcernerror-002.phpt [hunk 3]
```diff
@@ -23,7 +23,6 @@ try {
     $retval = $manager->executeBulkWrite(NS, $bulk, $w);
 } catch(MongoDB\Driver\Exception\BulkWriteException $e) {
     printWriteResult($e->getWriteResult(), false);
-    $e->getWriteResult(true);
```
Comment (derickr) [inline tests/replicaset/writeconcernerror-002.phpt]: Did this method go? If not, why remove it from the test?

File: tests/manager/manager-executeBulkWrite-004.phpt [hunk 4]
```diff
@@ -1,5 +1,5 @@
 --TEST--
-MongoDB\Driver\Manager::executeBulkWrite() with upserted ids
+MongoDB\Driver\Manager::executeBulkWrite() delete multiple documents
```
Comment (derickr) [inline tests/manager/manager-executeBulkWrite-004.phpt]: This test totally changes what it did before too?

File: tests/manager/manager-executeBulkWrite-002.phpt (hunk 1 above)
Comment (jmikola) [inline tests/manager/manager-executeBulkWrite-002.phpt]: The "duplicate key errors (ordered)" and "duplicate key errors (unordered)" tests were renamed to `manager-executeBulkWrite_error-001.phpt` and `manager-executeBulkWrite_error-002.phpt`, respectively. Git didn't pick up that it was a rename, though, as I ported over the single write method tests to `executeBulkWrite()` and kept the filename numbers in sequence.

File: tests/manager/manager-executeBulkWrite-004.phpt (hunk 4 above)
Comment (jmikola) [inline tests/manager/manager-executeBulkWrite-004.phpt]: `manager-executeBulkWrite-004.phpt` was essentially renamed to `manager-executeBulkWrite-002.phpt` after moving the duplicate key error tests. See https://github.com/mongodb-labs/mongo-php-driver-prototype/pull/128/files#r42785544.

File: tests/replicaset/writeconcernerror-002.phpt (hunk 3 above)
Comment (jmikola) [inline tests/replicaset/writeconcernerror-002.phpt]: I didn't see the point of this method call, as we're already calling it on the line above. Admittedly, this change should be in its own commit rather than lumped into "Rename WriteConcernError tests" (I can break it up if you like).

AFAIK, this was only testing that zpp expects no arguments, which didn't seem relevant for a WriteConcernError test.

File: tests/server/server-executeQuery-004.phpt (hunk 2 above)
Comment (jmikola) [inline tests/server/server-executeQuery-004.phpt]: The test is named `server-executeQuery-004` and yet its title was "Manager::executeDelete()" and it was only ever calling `Manager::executeInsert()` and `Manager::executeQuery()`. This all seemed very incorrect, so I fixed things to actually invoke the server methods.

`$manager->executeQuery(NS, new MongoDB\Driver\Query(array()))->getServer();` was something I borrowed from some other server tests. I suppose I could just have used `Manager::selectServer()` there to obtain the primary.

**Comment by derickr** [comment]
LGTM
